### PR TITLE
internal/integration: testscript for cli tests

### PR DESCRIPTION
**Pull request by rotemtam**

### CODE REVIEW

File: internal/integration/integration_test.go [hunk 1]
```diff
@@ -542,3 +541,11 @@ func (r *rrw) ReadRevisions(ctx context.Context) (migrate.Revisions, error) {
 func (r *rrw) clean() {
 	*r = rrw(migrate.Revisions{})
 }
+
+func compileCLI(t *testing.T) (string, error) {
```
Comment (a8m) [inline internal/integration/integration_test.go:545]:
```suggestion
func buildCmd(t *testing.T) (string, error) {
```
A suggestion, no need to rename if you don't like it.

File: internal/integration/script_test.go [hunk 2]
```diff
@@ -135,6 +137,11 @@ func (t *liteTest) setupScript(env *testscript.Env) error {
 	// environment as tests run in parallel.
 	env.Values[keyDB] = db
 	env.Values[keyDrv] = drv
+	cliPath, err := compileCLI(t.T)
```
Comment (a8m) [inline internal/integration/script_test.go:140]: Why not storing in the temporary directory (of testscript)?

File: internal/integration/script_test.go [hunk 3]
```diff
@@ -310,6 +317,31 @@ func (t *liteTest) cmdExec(ts *testscript.TestScript, _ bool, args []string) {
 	cmdExec(ts, args, ts.Value(keyDB).(*sql.DB))
 }
 
+func (t *liteTest) cmdCLI(ts *testscript.TestScript, _ bool, args []string) {
+	for i, arg := range args {
+		args[i] = strings.ReplaceAll(arg, "URL", t.dsn())
+	}
+	l := len(args)
+	switch {
```
Comment (a8m) [inline internal/integration/script_test.go:325]:
```suggestion
	switch l := len(args); {
```

File: internal/integration/script_test.go [hunk 4]
```diff
@@ -310,6 +317,31 @@ func (t *liteTest) cmdExec(ts *testscript.TestScript, _ bool, args []string) {
...
+			path := filepath.Join(wd, args[l-1])
+			out, err := os.Create(path)
+			if err != nil {
+				return err
+			}
+			defer out.Close()
+			cmd := exec.Command(ts.Getenv("cli"), args[0:l-2]...)
+			cmd.Stdout = out
+			return cmd.Run()
+		}())
```
Comment (a8m) [inline internal/integration/script_test.go:339]: why it's better than:
```suggestion
		wd := ts.Getenv("WORK")
		path := filepath.Join(wd, args[l-1])
		out, err := os.Create(path)
		ts.Check(err)
		defer out.Close()
		cmd := exec.Command(ts.Getenv("cli"), args[0:l-2]...)
		cmd.Stdout = out
		ts.Check(cmd.Run())
```

**Review by a8m — APPROVED** [review]
Looks good, added minor questions/suggestions

File: internal/integration/script_test.go (hunk 2 above)
Comment (rotemtam) [reply internal/integration/script_test.go]: dont want to compile it for every test case

**Pull request merged**